### ISSUE TITLE
Test channel buffered pings creates blocked fiber

### ISSUE THREAD

**Issue by wonderix** [crystal-lang/crystal#12135]
## Bug Report

The unit test ["buffered ping" in `channel_spec.cr`](https://github.com/crystal-lang/crystal/blob/9767d2110a34934787af795fd9f8e83d01b43105/spec/std/channel_spec.cr#L746-L751) is not correct. It creates a fiber, which is blocked because `ch.receive` in the main thread already consumes the item sent (before `ch.receive` is called inside `spawn`).

It should should be:

```crystal
  it "pings" do
    ch1 = Channel(Int32).new(10)
    ch2 = Channel(Int32).new(10)
    spawn { ch2.send(ch1.receive) }
    ch1.send 123
    ch2.receive.should eq(123)
  end
```

**Comment by yxhuvud** [comment]
Not certain the suggested new spec make a lot of sense either though, as it still passes without the buffer argument:
```
  ch1 = Channel(Int32).new
  ch2 = Channel(Int32).new
  spawn { ch2.send(ch1.receive) }
  ch1.send 123
  ch2.receive.should eq(123)
```
But perhaps that is fine?

**Comment by wonderix** [comment]
That's correct. In this case, I would move the `spawn` after the first `send`
